### PR TITLE
Project: Add 404 responses for projects and workflows

### LINKED ISSUE #1694: TypeError: Cannot read property 'active_workflows' of undefined
Sentry Issue: [FEM-APP-PROJECT-98](https://sentry.io/organizations/zooniverse-27/issues/1729391371/?referrer=github_integration)

```
TypeError: Cannot read property 'active_workflows' of undefined
  at HeroContainer_storeMapper (/usr/src/packages/app-project/.next/server/static/DebAPKhRjlAP6tKPCB3DQ/pages/projects/[owner]/[project].js:2004:48)
  at Injector.render (/usr/src/packages/app-project/node_modules/mobx-react/index.js:645:33)
  at Object.allowStateChanges [as _allowStateChanges] (/usr/src/packages/app-project/node_modules/mobx/lib/mobx.js:974:15)
  at None (/usr/src/packages/app-project/node_modules/mobx-react/index.js:851:28)
  at trackDerivedFunction (/usr/src/packages/app-project/node_modules/mobx/lib/mobx.js:764:24)
...
(23 additional frame(s) were not displayed)
```

We should not attempt to request for workflows when the project is not found and instead route to a 404 page with a user friendly error message.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Check the project model for a project ID after the project API call resolves. If there's no project ID, set a 404 status code and show the default Next error page.

When there's a workflow ID in the URL, return 404 if it isn't listed in the project's active workflows.

Split data-fetching, and page props that only depend on the page URL, into a `getStaticPageProps` helper.

Screenshot of a blank white error page, saying: 404, project btooke/i-fancy-cats was not found.

Package:
app-project

Closes #1694.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
I think this PR will 404 if Panoptes is down. Do we want to do that? The project model might need a bit of work to distinguish between API errors and successful requests for resources that don't exist.

**Comment by eatyourgreens** [comment]
The size of this PR jumped a lot today, but the increase is mostly tests. A lot of the SSR page props code has never had tests.

File: packages/app-project/stores/Project.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ const Project = types
     get defaultWorkflow() {
       const activeWorkflows = self.links['active_workflows']
       let singleActiveWorkflow
-      if (activeWorkflows.length === 1) {
+      if (activeWorkflows?.length === 1) {
```
Comment (srallen) [inline packages/app-project/stores/Project.js:44]: Is there a change that this would be undefined?

**Review by srallen — APPROVED** [review]
One minor style thing on the current generic 404 page:

Screen Shot 2021-02-12 at 2 28 52 PM

We have an extra period in the message. This is a non-issue since I think implementing @beckyrother's designs would be the next iteration? 

Nice set of new tests for this functionality with nock! Also, I since this partially implements what we discussed in #1997, it'd be nice to have the summary ADR of the decision written up to be able to reference while reviewing. Any chance you have time to write that up? Thanks!

File: packages/app-project/stores/Project.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-project/stores/Project.js:44]: It came up when I was first testing this branch but I don’t remember how.

Comment (srallen) [reply packages/app-project/stores/Project.js:44]: It seems really unlikely that Panoptes would not serialize this back to us, so likely has to do with how the project resource is being mocked for tests. Maybe we can setup rosie in the project app to consistent mocks for testing?

Comment (eatyourgreens) [reply packages/app-project/stores/Project.js:44]: It’s undefined for projects that don’t exist in Panoptes , but I don’t remember how I triggered that.

Comment (eatyourgreens) [reply packages/app-project/stores/Project.js:44]: Whatever that error was, I can't reproduce it either in the dev site or in the tests. I'm going to take this existence check out.

**Pull request merged**